ci/README.md

```diff
@@ -0,0 +1,43 @@
+# Developer use of CI Docker image
+
+The Docker image at `lyft/envoy-build:latest` is used for Travis CI checks. It
```

Comment:
small nit: as of https://github.com/lyft/envoy/pull/408 we are now pinning CI builds to a specific build image (easier to understand what is being tested, can upgrade an image and work on a PR, etc.). Do you mind just covering this briefly? (Basically that CI is now pinned, but latest will still work for dev if desired).